### PR TITLE
persist cookies; simplify API.

### PR DESCRIPTION
**Pull request by rogpeppe**
We make the default Options to persist using the standard public suffix list.
We lose the exported Merge, WriteTo and ReadFrom methods in favour
of just doing thing automatically.

We also make the merge work correctly when the public suffix list changes,
which requires changing the serialization format.

### CODE REVIEW

File: jar.go [hunk 1]
```diff
@@ -82,23 +85,49 @@ type Jar struct {
...
-var nextJarId uint64
+var noOptions Options
 
 // New returns a new cookie jar. A nil *Options is equivalent to a zero
 // Options.
+//
+// New will return an error if the cookies could not be loaded
+// from the file for any reason than if the file does not exist.
+//
+// Note that if o.Filename is empty, the cookie jar will not
```
Comment (mhilton) [inline jar.go]: This comment contradicts the other one.

File: jar_test.go [hunk 2]
```diff
@@ -6,15 +6,23 @@ package cookiejar
...
 	"sort"
 	"strings"
 	"sync"
+	"sync/atomic"
 	"testing"
 	"time"
 )
 
+// nonExistentFile holds a large random number that is vanishingly unlikely to exist for real.
+var nonExistentFile = "98818e574592c54bf1ec90174e72e33081a0f8324c6e6302fa94ff389e6b7fc8"
```
Comment (mhilton) [inline jar_test.go]: if you're worried you could create an init that panics if this file exists.

File: serialize.go [hunk 3]
```diff
@@ -6,42 +6,64 @@ package cookiejar
...
+	}
+	if _, err := f.Seek(0, 0); err != nil {
+		return errgo.Mask(err)
+	}
+	return j.writeTo(f)
 }
 
-// Load uses j.ReadFrom to read cookies
-// from the file at the given path. If the file does not exist,
+// load loads the cookies from j.filename. If the file does not exist,
```
Comment (mhilton) [inline serialize.go]: This comment could probably do with formatting.

**Comment by mhilton** [comment]
:+1: LGTM

**Comment by jrwren** [comment]
:+1:

File: jar.go (hunk 1 above)
Comment (rogpeppe) [inline jar.go]: fixed

File: jar_test.go (hunk 2 above)
Comment (rogpeppe) [inline jar_test.go]: i'm not worried :)

File: serialize.go (hunk 3 above)
Comment (rogpeppe) [inline serialize.go]: done

**Pull request merged**